datafusion/functions-array/src/lib.rs

```diff
@@ -98,54 +103,245 @@ pub mod expr_fn {
     pub use super::string::string_to_array;
 }
 
+
+
+pub type ScalarFactory = Box<dyn Fn() -> Arc<ScalarUDF> + Send + Sync>;
+
+/// HashMap Singleton for UDFs
+///
+/// Replace register_all with our built-in functions
+/// Replace  scalar_functions: HashMap<String, Arc<ScalarUDF>> in SessionState
+pub fn array_functions() -> &'static Mutex<HashMap<String, ScalarFactory>> {
+    static FUNCTIONS: OnceLock<Mutex<HashMap<String, ScalarFactory>>> = OnceLock::new();
+    FUNCTIONS.get_or_init(|| {
+        let mut functions = HashMap::new();
+        functions.insert(
+            String::from("array_to_string"),
```

Comment:
With alias support, this commit can run.
I think we need to public aliases function 🤔